source/developers-guide/plugin-configuration/index.md

```diff
@@ -0,0 +1,685 @@
+---
+layout: default
+title: Plugin configuration
+github_link: developers-guide/plugin-configuration/index.md
+indexed: true
+---
+This document will give you a brief introduction about how to set configuration options for plugins, which parameters are available and
+how to use them.
+
+<div class="toc-list"></div>
+
+## Introduction
+After getting a short introduction by reading the Plugin Quick Introduction, we will now be introduced to
+the configuration options for plugins. Shopware delivers a big amount of helper functions for generating standard
+configuration fields in backend.
+
+## Creating forms
+To create a simple plugin configuration we can use the function `Form()`
+
+```
+$form = $this->Form();
+```
+
+Now we can use this instance of `Shopware\Models\Config\Form` in `$form` to add elements to it via `setElement().
+```
+$form->setElement(type : String, name : String, [options : array | null= null]);
+```
+A textfield would be defined as followed:
+
+```
+public function createConfiguration()
+{
+    $form = $this->Form();
+ 
+    $form->setElement('text', 'text');
+}
+```
+
+Which would lead to a simple textfield.
+
+<img src="img/simple_textfield.png" alt="Simple textfield" class="image-border" />
+
+## Options Parameter
+The options parameter of the `setElement` function allows to set several configurations on the form element.
+
+### Label
+The label parameter allows to create a simple descriptional label for the form element.
+```
+public function createConfiguration()
+{
+    $form = $this->Form();
+ 
+    $form->setElement('text', 'text', 
+        array('label' => 'Text')
+    );
+}
+```
+
+<img src="img/textfield_label.png" alt="Textfield with label" class="image-border" />
+
+### Value
+The value parameter stands for the default value of the field if this hasn´t been edited yet. It will directly be shown in the configuration element.
+```
+public function createConfiguration()
+{
+    $form = $this->Form();
+ 
+    $form->setElement('text', 'text', 
+        array(
+            'label' => 'Text',
+            'value' => 'Vorauswahl'
+        )
+    );
+}
+```
+
+<img src="img/default_value.png" alt="Textfield with default value" class="image-border" />
+
+### Scope
+With help of the scope parameter it is possible to generate subshop specific configurations. If the scope parameter is not set,
+the configuration option is available for all subshops.
```

Comment:
i'm not sure, but this is wrong. if the scope is not set, you cannot configure this value for subshops, afaik